components/tools/OmeroJava/test/integration/ScriptServiceTest.java

```diff
@@ -192,18 +195,44 @@ public void testUploadOfficialScript() throws Exception {
     @Test
     public void testUploadOfficialScriptAsRoot() throws Exception {
         logRootIntoGroup();
-        StringBuffer buf = new StringBuffer("");
-        String[] values = { "a", "b", "c" };
-        for (int i = 0; i < values.length; i++) {
-            buf.append(values[i].charAt(0));
-        }
-        String folder = "officialTestFolder";
         IScriptPrx svc = factory.getScriptService();
-        try {
-            long id = svc.uploadOfficialScript(folder, buf.toString());
-            Assert.assertTrue(id > 0);
-        } catch (Exception e) {
-        }
+        List<OriginalFile> scripts = svc.getScripts();
+        OriginalFile f = scripts.get(0);
+        //read the script. This is tested elsewhere
+        String str = readScript(f);
+        String folder = f.getName().getValue();
+        long id = svc.uploadOfficialScript(folder, str);
+        Assert.assertTrue(id > 0);
+        Assert.assertEquals(svc.getScripts().size(), scripts.size()+1);
+        deleteScript(id);
+    }
+
+    /**
+     * Tests to upload an official script by a user who is an administrator,
+     * this method uses the <code>deleteScript</code>.
+     *
+     * @throws Exception
+     *             Thrown if an error occurred.
+     */
+    @Test
+    public void testDeleteScriptAsRoot() throws Exception {
+        logRootIntoGroup();
+        IScriptPrx svc = factory.getScriptService();
+        List<OriginalFile> scripts = svc.getScripts();
+        OriginalFile f = scripts.get(0);
+        //read the script. This is tested elsewhere
+        String str = readScript(f);
+        String folder = f.getName().getValue();
+        long id = svc.uploadOfficialScript(folder, str);
+        deleteScript(id);
+        Assert.assertEquals(svc.getScripts().size(), scripts.size());
+        //Check that the entry has been removed from DB
+        ParametersI param = new ParametersI();
+        param.map.put("id", omero.rtypes.rlong(id));
+        f = (OriginalFile) factory.getQueryService().findByQuery(
```

Comment:
`AbstractServerTest` has some useful `assertDoesNotExist` and suchlike that may help here.